proxy/context.go

```diff
@@ -284,7 +291,7 @@ func (c *context) Split() (filters.FilterContext, error) {
 	u.Host = originalRequest.Host
 	cr, body, err := cloneRequestForSplit(u, originalRequest)
 	if err != nil {
-		c.proxy.log.Errorf("context: failed to clone request: %v", err)
+		c.Logger(c.proxy.log).Errorf("context: failed to clone request: %v", err)
```

Comment:
This was we couple proxy logger with context logger.
I think we should add logger implementation that logs trace id and use it instead here:
https://github.com/zalando/skipper/blob/29c3ea1378faea74f7c19920ae2aeb341dd273c7/proxy/proxy.go#L715

Reply:
I see the problem now, we need to resolve trace id in runtime.
Maybe we can use `ctx.Logger()` instead of `c.proxy.log` where applicable?

Reply:
I tried this, see https://github.com/zalando/skipper/commit/76edb7929c0d4234538becb5c6ff0022d915b7cd

The reason why I haven't use such solution are tests failures, see https://github.com/zalando/skipper/actions/runs/5078553903/jobs/9123211440

The reason is that `make cicheck` reads information from `c.proxy.log` to test logging.

Reply:
Then maybe we should fix failing tests to not rely on proxy log?

Reply:
> Then maybe we should fix failing tests to not rely on proxy log?
When I have a look on failing tests, it's not obvious for me how to fix it because tests pass its own logger instance.
https://github.com/zalando/skipper/blob/v0.16.68/proxy/proxy_test.go#L162

However, as `proxy.context` has `proxy` field, I can get proxy log through it, without additional args.
Changed PR to do it.

Reply:
This would essentially make ctx.Logger() return proxy.log. I am not sure whether it is a good idea.
Is it hard to fix proxy tests that rely on proxy log? How many are there?

Reply:
6 tests. Is it hard?.. Maybe there is some easy way, but I don't see.

```
 func TestBackendTimeoutInTheMiddleOfServiceResponse(t *testing.T) {
 func TestClientTimeout(t *testing.T) {
 func TestClientCancellation(t *testing.T) {
 func TestClientTimeoutBeforeStreaming(t *testing.T) {
 func TestFilterPanic(t *testing.T) {
 func TestFilterPanicPrintStackRate(t *testing.T) {
```

Reply:
> This would essentially make ctx.Logger() return proxy.log

Actually I was thinking about it, and I still don't see any really convenient way to bind `loggingtest.Logger` (which is used by unit tests to count stats about logs and assigned to `c.proxy.log`) and `ctx.Logger()` which is created inside `proxy/proxy.go:ServeHttp` function.

Another solution which I could propose is to return proxy.log only if `ctx.Logger()` called inside unit tests.

This is what this code intended for:
```go
if c.proxy != nil && c.proxy.log != nil {
		switch c.proxy.log.(type) {
		case *loggingtest.Logger:
			c.logger = c.proxy.log
		}
	}
```

Reply:
Discussed with Alexander. It's better not to change production code to make tests happy. It must be vice versa.